// @harveyr/github-actions-kit/lib/github/checks.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Converts an annotation object from our internal interface to what the Github API expects.
 * @param d
 */
function convertAnnotation(d) {
    var title = d.title, level = d.level, message = d.message, path = d.path, startLine = d.startLine, endLine = d.endLine, startColumn = d.startColumn, endColumn = d.endColumn;
    return {
        title: title,
        // eslint-disable-next-line @typescript-eslint/camelcase
        annotation_level: level,
        // eslint-disable-next-line @typescript-eslint/camelcase
        start_line: startLine,
        // eslint-disable-next-line @typescript-eslint/camelcase
        end_line: endLine ? endLine : startLine,
        // eslint-disable-next-line @typescript-eslint/camelcase
        start_column: startColumn,
        // eslint-disable-next-line @typescript-eslint/camelcase
        end_column: endColumn,
        message: message,
        path: path,
    };
}
exports.convertAnnotation = convertAnnotation;
